### ISSUE TITLE
Exponentiation of a type with units works differently for `uint256` and `int128`

### ISSUE THREAD

**Issue by smarx** [ethereum/vyper#1704]
When the underlying type is `uint256`, all units are stripped away. When the underlying type is `int128`, the units remain but with an incorrect exponent (`m**3` rather than `m**6` below):

```python
units: {
    m: "meter"
}

@public
@constant
def test() -> int128(m**3):
    a: int128(m**2) = 5
    return a**3 # BUG: Type int128(m**3) instead of int128(m**6)

@public
@constant
def test2() -> uint256:
    a: uint256(m**2) = 5
    return a**2 # BUG: Type uint256 instead of uint256(m**4)
```

**Comment by charles-cooper** [comment]
should be fixed in https://github.com/ethereum/vyper/pull/1661

**Comment by charles-cooper** [comment]
confirmed fixed as of f7211c130a180156e64f7d6d141db30ef02e0c14. it's unclear to me if exponentiation should change the unit though, opening another issue for discussion.

```python
 $ vyper -f ir /dev/stdin <<EOF
> units: {
>     m: "meter"
> }
> 
> @public
> @constant
> def test() -> int128(m**3):
>     a: int128(m**2) = 5
>     return a**3 # BUG: Type int128(m**3) instead of int128(m**6)
> 
> @public
> @constant
> def test2() -> uint256:
>     a: uint256(m**2) = 5
>     return a**2 # BUG: Type uint256 instead of uint256(m**4)
> EOF
Error compiling: /dev/stdin
vyper.exceptions.TypeMismatchException: line 15:11 Trying to return base type uint256(m**2), output expecting uint256
     14     a: uint256(m**2) = 5
---> 15     return a**2 # BUG: Type uint256 instead of uint256(m**4)
-------------------^
     16
```
